Test/git-issues/git-issue-1479a.dfy

```diff
@@ -0,0 +1,14 @@
+// RUN: %dafny /compile:0 "%s" > "%t"
+// RUN: %diff "%s.expect" "%t"
+
+type T
+datatype BaseType = BaseType(t: T)
+
+predicate P(d: BaseType)
+type SubsetType = d: BaseType | P(d) witness *
+
+method m0(dp: SubsetType)
+
+method m1(dp: SubsetType, t: T) {
+  var dp': SubsetType := [dp][0].(t := t);
```

Comment:
Add a comment here, saying we expect an error on this line.

Reply:
```suggestion
  var dp': SubsetType := [dp][0].(t := t); // Error: `t` does not satisfy subtype constraints
```